### ISSUE TITLE
uv pip install --target and workspace

### ISSUE THREAD

**Issue by jpambrun-vida** [astral-sh/uv#5631]
I am struggling to understand the expected behavior of `uv pip install --target /dest` in conjunction with workspaces. 

with 
```
[tool.uv.sources]
lambda_utils = { workspace = true }
```

I want to use --target in my docker build step, but it will only create a `.pth`. If I add  `editable = false`, `-- target` works as expect, but then I loose linking with developing. Shouldn't --target always copy?

**Comment by charliermarsh** [comment]
Yeah this makes sense as a problem. There are two directions from which we could solve it: (1) adding some kind of `--no-editable` flag to `uv pip install`, or (2) adding a more first-class `--target` interface to `uv sync`.

**Comment by charliermarsh** [comment]
I'm more inclined to do (2).

**Comment by dmbonsall** [comment]
I personally like the idea of adding some sort of `--target` flag to `uv sync` because it would allow the package versions to be tied to the lock-file. I'd like to use this to create a directory I can zip up and upload to a lambda, so ensuring the dependency versions are the same everywhere would be great.

My only ask would be that the specified `--target` directory would be fully standalone so it can ship as a single unit.

**Comment by Sillocan** [comment]
I made a reproducible script for this issue:

```sh
cd $(mktemp -d)
uv init

# setup library in the workspace
uv init --lib library
uv add library

# setup the lambda function referencing the library in the workspace above
uv init --lib lambda_function
uv add lambda_function
echo "def handler(): ..." > lambda_function/src/lambda_function/handler.py
uv add --directory="lambda_function" requests ../library

# update the lock file
uv sync --all-packages

# export the requirements file with `--no-editable` specified
uv export \
    --frozen \
    --no-dev \
    --no-editable \
    --package="lambda_function" \
    -o requirements.txt

uv pip install \
    --no-installer-metadata \
    --no-compile-bytecode \
    --python-platform x86_64-manylinux2014 \
    --python 3.13 \
    --target "_lambda_build" \
    -r requirements.txt
ls _lambda_build
```


The output above is that `lambda_function` is correct, but `library` is editable.
```sh
$ ls _lambda_build
bin
certifi
certifi-2025.1.31.dist-info
charset_normalizer
charset_normalizer-3.4.1.dist-info
idna
idna-3.10.dist-info
lambda_function
lambda_function-0.1.0.dist-info
library-0.1.0.dist-info
_library.pth
requests
requests-2.32.3.dist-info
urllib3
urllib3-2.3.0.dist-info
```

**Comment by AndyReifman** [comment]
I'm running into this issue as well.

Has anyone been able to find a good workaround in the meantime while the functionality doesn't exist?

**Comment by zanieb** [comment]
Does `--no-sources` work?

**Comment by AndyReifman** [comment]
Hmm. Not sure if it's a lack of uv knowledge or an actual issue but I'm hitting an error trying to resolve dependencies with `--no-sources`

Basic outline of the project is

```toml
# pyproject.toml
# ...
[tool.uv.workspace]
members = [
    "packages/*"
]

# packages/package_a/pyproject.toml
# ...
[tool.uv.sources]
"package_b" = { workspace = true } 

# packages/package_c/pyproject.toml
# ...
[tool.uv.sources]
"package_b" = { workspace = true }
```

Run with the command
`uv export --no-dev --no-editable --no-sources --package="package-a" -o requirements.txt`

The error I'm getting 

```
No solution found when resolving dependencies:
    Because package_c depends on package_b and your workspace requires package_c, we can conclude that your workspace's requirements are unsatisfiable"
```

I'm not even sure why it's trying to resolve dependencies for `package_c` as it's not used at all in `package-a`

**Comment by zanieb** [comment]
We solve for the entire workspace to generate the `uv.lock` regardless of the selected subset of packages.

In this case, I'm suggesting `--no-sources` on the `uv pip install` invocation. You'll need sources to load your workspace members from the local source tree.

**Comment by AndyReifman** [comment]
Ah. That does look to be working! 

Thanks! I wasted way too much time trying to get that work previously.

**Comment by zanieb** [comment]
Ah great, glad that works. This is roughly the problem discussed in https://github.com/astral-sh/uv/issues/14617 as well.

**Comment by jtdb-pariveda** [comment]
bumping this, what is the solution in the context of exporting some package to bundle into a lambda?
in particular, I'm personally interested in bundling all DEPENDENCIES of my lambda app, including local libraries, which get uploaded to a lambda layer... but the same question still applies if trying to bundle it all together for a lambda zip with all dependencies including a workspace library.

I've confirmed that you can do it from the project root by doing something like:
```
uv export --no-emit-project --project=my_project --no-dev --format requirements-txt > my-project/test-build/requirements.txt

uv pip install \
    --requirement my-project/test-build/requirements.txt \
    --target my-project/test-build/common_layer/python \
    --python-platform x86_64-manylinux_2_28 \
    --no-deps
```

Like, the output, whether running from workspace root or from within a workspace project is always going to be:
```
-e ./shared-lib
    # via my_project
```

But I'd like to be able to run this from within my workspace project for the lambda, and be able to resolve the exported requirements.txt with `../shared-lib` (correct vs project) instead of `./shared-lib` which is incorrect vs. project and correct vs, workspace root.
